az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -581,12 +583,22 @@ public static String javaOptStringFromHadoopConfiguration(Configuration conf, St
    * @return a CSV of tags
    */
   public static String constructHadoopTags(Props props, String[] keys) {
-    String[] keysAndValues = new String[keys.length];
+    String[] keysAndValues = new String[keys.length + 1];
     for (int i = 0; i < keys.length; i++) {
       if (props.containsKey(keys[i])) {
         keysAndValues[i] = keys[i] + ":" + props.get(keys[i]);
       }
     }
+    if (props.containsKey(CommonJobProperties.PROJECT_NAME)
+        && props.containsKey(CommonJobProperties.FLOW_ID)) {
+      String constructedTag = "workflowid:"
+          + props.get(CommonJobProperties.PROJECT_NAME)
+          + HadoopConfigurationInjector.WORKFLOW_ID_SEPERATOR
+          + props.get(CommonJobProperties.FLOW_ID);
+      constructedTag = constructedTag.substring(0,
+          Math.min(constructedTag.length(), APPLICATION_TAG_MAX_LENGTH));
+      keysAndValues[keys.length] = constructedTag;
```

Comment:
Does it matter that both the PROJECT_NAME and FLOW_ID are already included in the output, individually, since the three keys passed here are always:
        CommonJobProperties.EXEC_ID,
        CommonJobProperties.FLOW_ID,
        CommonJobProperties.PROJECT_NAME
Do we have to repeat this info one more time as a combined tag?

Reply:
Hi @li-ygerchikov , yeah, we need a combined tag for this. The API in Hadoop 2.10 requires a tag of the form "workflowid:...", it is not able to combine the existing tags on the server side (the project_name and flow_id tags are azkaban specific anyway), hence we combine it here.

Reply:
OK, that makes sense, and it helps -- thanks.
The only other consideration would be whether we'd like to keep this *util* method completely generic, and instead add the specific tag that you need in setupHadoopJobProperties in azkaban/jobtype/AbstractHadoopJavaProcessJob.java?

Reply:
@li-ygerchikov  that makes sense. Updated the PR.